### PR TITLE
update plugin dependencies

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Updated build process for the plugin.
Updated backend plugin dependencies
Added check via the Grafana check tool

### CODE REVIEW

**Comment by hagen1778** [comment]
Why do we need magefile.go?

File: deployment/docker/Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ PKG_PREFIX := github.com/VictoriaMetrics/victorialogs-datasource
 
 ROOT_IMAGE ?= alpine:3.20.0
 CERTS_IMAGE := alpine:3.20.0
```
Comment (AndrewChubatiuk) [inline deployment/docker/Makefile:5]: please upgrade these images as well

File: go.mod [hunk 2]
```diff
@@ -1,99 +1,100 @@
 module github.com/VictoriaMetrics/victorialogs-datasource
 
-go 1.21.6
+go 1.23.1
```
Comment (AndrewChubatiuk) [inline go.mod:3]: please upgrade ci jobs as well

Comment (AndrewChubatiuk) [reply go.mod:3]: or just set it in CI jobs like it's described here
https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file

Comment (dmitryk-dk) [reply go.mod:3]: fixed

File: deployment/docker/Makefile (hunk 1 above)
Comment (dmitryk-dk) [reply deployment/docker/Makefile]: fixed

**Comment by dmitryk-dk** [comment]
> Why do we need magefile.go?

we updated the build process

**Review by hagen1778 — APPROVED** [review]

**Review by AndrewChubatiuk — DISMISSED** [review]

**Review by AndrewChubatiuk — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,8 +2,12 @@
 
 ## tip
 
+
+* FEATURE: update plugin dependencies and add a check process for all releases the integrity of plugin.
```
Comment (hagen1778) [inline CHANGELOG.md:6]:
```suggestion
* FEATURE: update plugin dependencies to satisfy Grafana marketplace requirements
* FEATURE: add check process for plungi integrity on releases.
```

**Review by hagen1778 — DISMISSED** [review]

**Review by hagen1778 — APPROVED** [review]

**Pull request merged**